mono/metadata/domain.c

```diff
@@ -1196,8 +1196,20 @@ mono_domain_free (MonoDomain *domain, gboolean force)
 	if (domain->aot_modules)
 		mono_jit_info_table_free (domain->aot_modules);
 	g_assert (domain->num_jit_info_table_duplicates == 0);
-	mono_jit_info_table_free (domain->jit_info_table);
-	domain->jit_info_table = NULL;
+
+	while (TRUE) {
```

Comment:
Why is this needed ? When the domain is freed, there should be nobody else accessing this table.

Reply:
Crashes observed that implied data races around this pointer / were fixed by this. A memory barrier didn't seem to do the trick. I can place this code in another PR and we can wait to see if we get any flakes/crashes without it on CI, if you'd prefer.